units/util.go

```diff
@@ -30,32 +30,42 @@ func HandlePoisonedHost(ctx context.Context, env evergreen.Environment, h *host.
 			}
 
 			for i := range containers {
-				catcher.Wrapf(DisableAndNotifyPoisonedHost(ctx, env, &containers[i], reason), "disabling poisoned container '%s' under parent '%s'", containers[i].Id, h.ParentID)
+				catcher.Wrapf(DisableAndNotifyPoisonedHost(ctx, env, &containers[i], true, reason), "disabling poisoned container '%s' under parent '%s'", containers[i].Id, h.ParentID)
 			}
-			catcher.Wrapf(DisableAndNotifyPoisonedHost(ctx, env, parent, reason), "disabling poisoned parent '%s' of container '%s'", h.ParentID, h.Id)
+			catcher.Wrapf(DisableAndNotifyPoisonedHost(ctx, env, parent, true, reason), "disabling poisoned parent '%s' of container '%s'", h.ParentID, h.Id)
 		}
 	} else {
-		catcher.Wrapf(DisableAndNotifyPoisonedHost(ctx, env, h, reason), "disabling poisoned host '%s'", h.Id)
+		catcher.Wrapf(DisableAndNotifyPoisonedHost(ctx, env, h, true, reason), "disabling poisoned host '%s'", h.Id)
 	}
 
 	return catcher.Resolve()
 }
 
-func DisableAndNotifyPoisonedHost(ctx context.Context, env evergreen.Environment, h *host.Host, reason string) error {
+// DisableAndNotifyPoisonedHost disables an unhealthy host so that it cannot run
+// any more tasks, clears any tasks that have been straned on it, and enqueues a
+// job to notify that a host was disabled. If canDecommission is true and the
+// host is an ephemeral host, it will decommission the host instead of
+// quarantine it.
+func DisableAndNotifyPoisonedHost(ctx context.Context, env evergreen.Environment, h *host.Host, canDecommission bool, reason string) error {
```

Comment:
Can we make this function no-op if the task is already quarantined or decommissioned depending on the case?

Reply:
L50-52 should already no-op when the host is unhealthy. Or did you mean something about the task?

Reply:
Yeah disregard